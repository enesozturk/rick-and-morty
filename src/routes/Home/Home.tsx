import React, { FunctionComponent } from 'react';

// Components
import CharacterList from '../../components/CharacterList';
import Typography from '@material-ui/core/Typography';

// Styles
import './Home.scss';

const Home: FunctionComponent = () => {
  return (
    <>
      <Typography variant="h1" gutterBottom>
        Rick and Morty
      </Typography>
      <CharacterList />
    </>
  );
};

export default Home;
