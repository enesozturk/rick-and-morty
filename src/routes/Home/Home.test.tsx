import React from 'react';
import renderer from 'react-test-renderer';

import Home from './Home';

describe('Home Screen', () => {
  it('Renders as expected', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
