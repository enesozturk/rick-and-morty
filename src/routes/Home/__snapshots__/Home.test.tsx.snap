// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen Renders as expected 1`] = `
Array [
  <h1
    className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom"
  >
    Rick and Morty
  </h1>,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  >
    <div
      className="character-list-loading"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;
